stable/database/templates/service-headless.yaml

```diff
@@ -1,4 +1,5 @@
 {{- if or (not (hasKey .Values.database.te.dbServices "enabled")) (eq (include "defaultfalse" .Values.database.te.dbServices.enabled) "true") }}
```

Comment:
Does this Service do anything? It is supposed to select TEs, but isn't the point of a Headless Service to allocate DNS entries, which is impossible with a Deployment?

Reply:
Correct! There is no point in keeping this service. Maybe WeLab still use it to connect to the database using `direct=true`. @acabrele Do you know if they changed to using the clusterip service?

I wanted to remove it but it will be in a separate PR but I'm happy to hear other thoughts if it is used in any way.

Reply:
This was in use from before we had a cluster service available.  The headless service can have issues with load balancing too as it is dependant on DNS config.  I support removing in another PR once confirming it is not used by anyone.